diff -Nur kaffeine-0.8.1/kaffeine/src/player-parts/kaffeine-part/kaffeine_part.cpp kaffeine-0.8.1.modif/kaffeine/src/player-parts/kaffeine-part/kaffeine_part.cpp
--- kaffeine-0.8.1/kaffeine/src/player-parts/kaffeine-part/kaffeine_part.cpp	2006-04-03 00:48:25.000000000 +0200
+++ kaffeine-0.8.1.modif/kaffeine/src/player-parts/kaffeine-part/kaffeine_part.cpp	2006-09-02 00:23:53.000000000 +0200
@@ -1448,6 +1448,9 @@
 
 void KaffeinePart::saveConfig()
 {
+	if(!m_imageWindow) // m_audioVisual not initialize
+		return;
+
 	if (!m_audioVisual->items().count()) // no config loaded
 		return;
 
