Index: kaffeine/kaffeineui.rc
===================================================================
--- kaffeine/kaffeineui.rc	(revision 46)
+++ kaffeine/kaffeineui.rc	(working copy)
@@ -80,6 +80,7 @@
 </ToolBar>
 <ToolBar name="playlistToolbar"><text>Playlist Toolbar</text>
   <Action name="playlist_new"/>
+  <Action name="playlist_remove"/>
   <Action name="playlist_selector"/>
 </ToolBar>
 <State name="no_media_part">
