Index: kaffeine/player-parts/gstreamer-part/gstreamer_part.desktop
===================================================================
--- kaffeine/player-parts/gstreamer-part/gstreamer_part.desktop	(revision 46)
+++ kaffeine/player-parts/gstreamer-part/gstreamer_part.desktop	(working copy)
@@ -1,11 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
 Icon=gstreamer
 Name=GStreamerPart
 Comment=A player part based on GStreamer.
 Comment[de]=Ein auf GStreamer basierender player part.
-MimeType=application/x-ogg;audio/x-mp3;audio/x-mpeg;audio/x-ogg;audio/x-wav;audio/x-flac;video/x-matroska;audio/x-matroska;video/mpeg;video/x-avi;application/x-kaffeine
+MimeType=application/x-ogg;audio/basic;audio/vnd.rn-realaudio;audio/x-aiff;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-wma;audio/x-ogg;audio/x-pn-realaudio;audio/x-pn-realaudio-plugin;audio/x-scpls;audio/x-wav;audio/x-flac;video/x-matroska;audio/x-matroska;video/mpeg;video/msvideo;video/quicktime;video/vnd.rn-realvideo;video/x-avi;video/x-fli;video/x-flic;video/x-ms-asf;video/x-ms-wmv;video/x-msvideo;application/x-mplayer2;application/smil;application/x-kaffeine;audio/x-musepack;
 ServiceTypes=KParts/ReadOnlyPart,KMediaPart
 Type=Service
 X-KDE-Library=libgstreamerpart
-InitialPreference=1
\ No newline at end of file
+InitialPreference=1
+
