Index: kaffeine/player-parts/kaffeine-part/kxinewidget.cpp
===================================================================
--- kaffeine/player-parts/kaffeine-part/kxinewidget.cpp	(revision 46)
+++ kaffeine/player-parts/kaffeine-part/kxinewidget.cpp	(working copy)
@@ -2689,13 +2689,7 @@
 	if ( m_lengthInfoTimer.isActive() ) m_lengthInfoTimer.stop();
 	//emit signalNewPosition(0, QTime());
 
-	if ((m_logoFile.isNull()) && (isPlaying()))
-		xine_stop(m_xineStream);
-	else
-	{
-		appendToQueue(m_logoFile);
-		QTimer::singleShot(0, this, SLOT(slotPlay()));
-	}
+	xine_stop(m_xineStream);
 
 	emit signalXineStatus(i18n("Stop"));
 }
